corehq/apps/accounting/bootstrap/utils.py

```diff
@@ -208,6 +162,69 @@ def _ensure_feature_rates(feature_rates, features, edition, verbose, apps):
     return db_feature_rates
 
 
+def _ensure_software_plan(plan_key, product, product_rate, verbose, apps):
+    SoftwarePlan = apps.get_model('accounting', 'SoftwarePlan')
+
+    plan_opts = {
+        'name': _software_plan_name(product, product_rate, plan_key.is_trial, plan_key.is_report_builder_enabled),
+        'edition': plan_key.edition,
+        'visibility': SoftwarePlanVisibility.PUBLIC
+    }
+    software_plan, created = SoftwarePlan.objects.get_or_create(**plan_opts)
```

Comment:
That's a good point, and I can't tell which is better here.

Previously, we would instantiate a new object with these parameters, then try to look one up in the db based on the name. If we find it based on its name, which is required to be unique between plans, we're basically throwing away `edition` and `visibility`. If we don't find it based on its name, we create a new plan using all three parameters.

Now, we will try to find an object in the db based on all three parameters, and if it doesn't match we create a new plan using the parameters. If the new object tries to use the same name as an existing plan, we will get an error, because name must be unique.

The only practical difference in behavior I see is if, for some reason we pass in a plan that has the same name but a different `edition` and/or `visibility` than an existing plan. Previously, we'd silently ignore those keys. Now, we'll get an error that name must be unique.

Reply:
On second thought, while I like the cleanness of `get_or_create` I think it's unnecessary to construct and pass in all the options here. Since we already know the name must be unique, we can try to `get` the plan first, then if that fails by `DoesNotExist`, simply `create` the new plan. That will retain existing behavior while structuring to be a little neater.